Sources/UInt128.swift

```diff
@@ -806,12 +679,31 @@ extension UnsignedInteger {
 // MARK: - Extend String for UInt128
 extension String {
     public init(_ value: UInt128) {
-        self.init()
-        self.append(value.toString())
+        self = value.toString()
     }
     public init(_ value: UInt128, radix: Int, uppercase: Bool = true) {
-        self.init()
-        let string = value.toString(radix: radix, uppercase: uppercase)
-        self.append(string)
+        self = value.toString(radix: radix, uppercase: uppercase)
+    }
+
+    var radix : UInt8 {
+        if hasPrefix("0b") { // binary
+            return 2
+        } else if hasPrefix("0o") { // octal
+            return 8
+        } else if hasPrefix("0x") { // hex
+            return 16
+        } else { // default to decimal.
+            return 10
+        }
+    }
+}
+
+extension UInt64 {
+    var byteSwapped : UInt64 {
```

Comment:
👎 

It turns out that UInt64 already has this logic built in with Swift 3.1, so there is no need to re-invent that wheel. It's called by the same name even!